### PR TITLE
darwin: fix uv_barrier race condition

### PR DESCRIPTION
**Pull request by cabelitos**
Prior to this patch a race condition may occur
in case a thread tries to destroy the barrier
while other awaiting threads were not
notified. Since the internal mutex and
condition variables are destroyed this may
cause an undefined behavior as described
by the man pages.
So in order to prevent such scenarios the
detroy function will not wait until all
awaiting threads are finished before
proceeding.

Closes: #3102

### CODE REVIEW

File: src/unix/thread.c [hunk 1]
```diff
@@ -122,7 +121,8 @@ void uv_barrier_destroy(uv_barrier_t* barrier) {
   uv_mutex_lock(&b->mutex);
 
   assert(b->in == 0);
-  assert(b->out == 0);
+  while (b->out != 0)
+    uv_cond_wait(&b->cond, &b->mutex);
 
   if (b->in != 0 || b->out != 0)
```
Comment (vtjnash) [inline src/unix/thread.c:127]:
```suggestion
  if (b->in != 0)
```

Comment (cabelitos) [reply src/unix/thread.c:127]: ty for the review. I update the PR with the proposed changes.

**Comment by cabelitos** [comment]
Hey guys, I updated the code.

**Review by vtjnash — APPROVED** [review]
Is it feasible to test this?

**Comment by vtjnash** [comment]
CI: https://ci.nodejs.org/job/libuv-test-commit/2179/

**Comment by cabelitos** [comment]
thank you!

**Pull request merged**